src/utilities/dicomJson.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Converts a naturalized PersonName (json) to a part10 PersonName (string)
+ * @param {object} value json-style person name
+ * @returns {string} part10 dicom representation of PersonName
+ */
+function pnDenaturalize(value) {
+    if (value && typeof value === "object") {
+        return `${value.Alphabetic ?? ""}=${value.Ideographic ?? ""}=${
+            value.Phonetic ?? ""
+        }`.replace(/=*$/, "");
+    }
+    return value;
+}
+
+/**
+ * Overrides toJSON to ensure serialization from part10 to json outputs the
+ * correct structure
+ * @param {string} tag part10 PersonName (PN) tag
+ */
+function pnFromPart10(tag) {
+    // Signal that this value was deserialized from a dcm/part10 file
+    tag.values.__pnDcm = true;
+    // To ensure proper json output, this value will be treated as the json dicom
+    // equivalent when serializing.
+    tag.values.toJSON = function () {
+        const components = this[0]?.split("=");
+        if (components && components.length) {
+            return [
+                {
+                    Alphabetic: components[0],
+                    Ideographic: components[1],
+                    Phonetic: components[2]
+                }
+            ];
+        }
+        return [];
+    };
+}
+
+/**
+ * Naturalizes a PN loaded from any source (part10 or json) by adding
+ * accessors to the name components, or overriding toString to allow
+ * denaturalization. See the note in the function for further details.
+ * @param {object} data object or string to be normalized to dicom json
+ */
+function pnNormalizeDicomToJson(data) {
+    // "PN" is one of the few deviations from the binary model, see:
+    // https://dicom.nema.org/dicom/2013/output/chtml/part18/sect_F.2.html
+
+    // If the PN object was created from a dcm file, it will be a ["string"].
+    // In this case ValueRepresentation will add a toJSON overload to the
+    // containing array object. If the object was created from JSON (by
+    // DICOMWeb for instance), toJSON should work as expected, and the
+    // denaturalization process will handle the converstion.
+
+    // This following code provides a consistent accessor experience for
+    // the naturalized dataset. If the value is a string, it mocks a json
+    // object, and if it's a json object it mocks a string by overriding
+    // toString. The latter ensures the ValueRepresentation output is
+    // correct.
+    if (
+        data.Value.__pnDcm ||
+        (data.Value &&
+            Array.isArray(data.Value) &&
```

Comment:
This doesn't deal with tags such as ConsultingPhysician's Name which can contain 1...n instances.  I know that is really a pain to have to deal with, as it will have to be decided how that works, but my suggestion is to:
1. If the VM !== 1 or 0 or 1, then leave the object as an array of objects all the time.
2. For other VM sizes, apply the change to the individual strings in each item, as a new String(originalValue)

Reply:
Oh my. I indeed didn't know PN could have VM > 1. I'll do another pass with that in mind, thanks for the heads up.